databuilder/codes.py

```diff
@@ -110,7 +110,17 @@ class DMDCode(BaseCode):
 def codelist_from_csv(filename, *, column, category_column=None):
     filename = Path(filename)
     if not filename.exists():
-        raise CodelistError(f"No CSV file at {filename}")
+        filename = str(filename)
+        if "\\" in filename:
```

Comment:
I feel like I'm missing something.  Why are we looking for double backslashes?
Errors like [this one](https://github.com/opensafely/openprompt_health_utilisation/blob/c862c96a5d9216c7376f4e75eebb135082c104f2/analysis/codelists.py#L52) result from a user trying to pass a string with a single backslash, like `codelists\opensafely-lung-cancer.csv`.

I'd assumed that calling Path on the filename was escaping the `\` into `\\`, but in a shell I get:
```
>>> filename = "foo\foo.csv"
>>> Path(filename)
PosixPath('foo\x0coo.csv')
>>> str(Path(filename))
'foo\x0coo.csv'
```

Reply:
:facepalm: I was using double-backslash as I've got used to that being how you spell backslash in Python e.g.
```
>>> print("\\")
\
```
But of course the problem is the user's backslashes aren't even being interpreted as backslashes — they're treated as escape sequences.

Thanks for catching this. Back to the drawing board!

Reply:
Aah! good detective work!